### PR TITLE
Convert asmamdgpu to ts

### PR DESCRIPTION
**Pull request by mattgodbolt**
Feel free to merge if you approve :)

### CODE REVIEW

File: lib/parsers/asm-parser.ts [hunk 1]
```diff
@@ -80,7 +80,7 @@ export class AsmParser extends AsmRegex implements IAsmParser {
     stdInLooking: RegExp;
     endBlock: RegExp;
     blockComments: RegExp;
-    constructor(compilerProps) {
+    constructor(compilerProps?) {
```
Comment (matsjla) [inline lib/parsers/asm-parser.ts:83]: We can type CompilerProps as PropertyGetter

**Review by matsjla — COMMENTED** [review]
Is the asm-parser-amdgpu file not supposed to have any changes (apart from rename?)

File: lib/parsers/asm-parser.ts (hunk 1 above)
Comment (mattgodbolt) [reply lib/parsers/asm-parser.ts:83]: Great point! I'll rebase post jeremy's big compiler thing and update.

**Comment by mattgodbolt** [comment]
done, ptal! :)

**Comment by mattgodbolt** [comment]
well, that broke

**Pull request merged**